### ISSUE TITLE
Manifest v2 support to be deprecated in Jan 2023

### ISSUE THREAD

**Issue by Jon8RFC** [warren-bank/crx-googlesource-raw-format#3]
I saw the error in the extension details, if you were unaware
https://developer.chrome.com/blog/mv2-transition/

**Thank you, so much, for creating this.**
I am not a linux guy, by any means.  Not having to figure out how to navigate 12gigs of git unpacked, nondescript files/folder-trees after git cloning just for two small boot.img files?  This extension solved that junk.

Maybe everyone knows, and maybe some just gave up.  I had to unzip the extension file and then go to "load unpacked" with developer mode enabled in the extensions window, to be able to enable the extension.  That may be handy info to list.

I saw in a closed issue that you'd push an update for an icon--are you able to list the extension on the chrome store?

Actually, is it against the TOS for you to make a pseudo-dummy extension which actually DOES do something, like display the latest version of your extension on github?  Meanwhile, the real purpose of the extension is just for you to link people to here, and here is were you have instructions on how to install the extension.  That way you'd be back in the chrome store, and more easily found.

**Comment by warren-bank** [comment]
Thanks for the link.. I wasn't aware that Chromium extensions having a manifest in v2 format would be entirely disabled in Jan 2023.

With regard to what you wrote:
* you probably had to unpack the extension..
  - not because of the version of its manifest,
  - but the fact that Chrome (not Chromium) is a walled garden, and only allows .crx extensions from the Play Store;
  - with the one exception that it will install unpacked extensions, which (by definition) cannot be signed
* personally, I don't use Chrome (proper)..
  - I like using forks of Chromium (ex: SRWare Iron) that remove all of the ties to Google services
  - which includes removing the restriction that .crx extensions need to be signed only by Google
* another option for Chrome users..
  - many extensions are little more than userscripts that modify the behavior of a single website,
  - which is the case for this particular repo,
  - which is exactly the reason that I added [a branch](https://github.com/warren-bank/crx-googlesource-raw-format/tree/greasemonkey-userscript) and released this as a Greasemonkey/Tampermonkey/WebMonkey [userscript](https://github.com/warren-bank/crx-googlesource-raw-format/raw/greasemonkey-userscript/greasemonkey-userscript/googlesource-raw-format.user.js)
    * so even though I don't publish this extension in the Play Store.. those userscript engines are available through the "official" channel (ie: within the walled garden)
    * and once installed, they can run any userscripts that you wish to add
* as to the question of why I don't publish my Chromium extensions (or Android apps) to the Google Play store..
  1. I don't want the headache of dealing with the bureaucracy of getting each update to pass through the review process
     - I learned that lesson with Firefox AMO.. never again
  3. I don't want to pay Google a fee for them to publish a link to my unpaid/free open-source work..
     - from which they earn ad revenue
     - and perpetuate the existence of a walled garden that I believe should not exist

tl;dr:
* moving forward, any Chromium extension that I post/update will use v3 of the manifest format
* for this utility, you should probably just add this Greasemonkey/Tampermonkey/WebMonkey [userscript](https://github.com/warren-bank/crx-googlesource-raw-format/raw/greasemonkey-userscript/greasemonkey-userscript/googlesource-raw-format.user.js)

**Comment by warren-bank** [comment]
PS: This extension/userscript is very handly for browsing small/medium size files.. but I'd still recommend using the command-line for downloading larger files. Strictly speaking, where you draw the line should only depend on how much RAM you have; though I tend use a cutoff on the order of MBs, rather than GBs.. since it's only a little extra effort, and a lot more efficient.

ex:
```bash
curl 'https://android.googlesource.com/platform/external/chromium-webview/+/refs/tags/android-7.1.2_r39/prebuilt/arm64/webview.apk?format=TEXT' | base64 --decode > 'com.android.webview_52.0.2743.100_arm8.apk'
```

**Comment by Jon8RFC** [comment]
That's a lot of great info...and laid out very nicely!

I really appreciate the alternatives you mentioned, especially the *monkey options.  I've recently started using some scripts again through that venue, so getting that might play more nicely if Chrome gives me a hard time.

I'm curious to try some non-Microsoft chromium-based browsers sometime, too.

Yep.  I've heard of the nightmares on Android/Play Store of great apps getting wrongly rejected for irrelevant reasons, or taking too long in the review process for updates.

...And nonsense like how LG, among many others, are permitted to use code to abuse root detection and disallow functionality on rooted devices.  Meanwhile Honeywell allows Alexa to manage my thermostat, so anyone walking by an open window, or just talking loudly enough, can change it.  But, should my rooted phone somehow manage to raise the temperature one degree without my authorization...that's something Honeywell doesn't want and Google permits.
Meanwhile, malware slips through regularly. But someone being able to see when the garbage truck is in their neighborhood, get a notification that their LG dryer finished, or change their thermostat is something that is too dangerous to be permitted on a rooted device "because. security" and Google allows it!

**Comment by Jon8RFC** [comment]
And, thanks for the detailed response.  I appreciated that.
